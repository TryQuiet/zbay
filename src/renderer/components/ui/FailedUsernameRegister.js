import React from 'react'
import PropTypes from 'prop-types'
import * as R from 'ramda'
import Grid from '@material-ui/core/Grid'
import Typography from '@material-ui/core/Typography'
import { withStyles } from '@material-ui/core/styles'

import red from '@material-ui/core/colors/red'

import Icon from './Icon'
import exclamationMark from '../../static/images/exclamationMark.svg'
import Modal from './Modal'
import LoadingButton from './LoadingButton'

const styles = theme => ({
  root: {
    padding: theme.spacing(4)
  },
  icon: {
    fontSize: '10rem',
    color: red[500],
    width: 80,
    height: 70
  },
  message: {
    wordBreak: 'break-all',
    marginTop: 36,
    fontWeight: 500
  },
  info: {
    textAlign: 'center'
  },
  button: {
    textTransform: 'none',
    width: 187,
    height: 60,
    fontSize: 16,
    lineHeight: '19px',
    fontWeight: 'normal',
    marginTop: 24
  }
})

export const FailedUsernameRegister = ({
  classes,
  open,
  handleClose,
  openModalCreateUsername
}) => {
  return (
    <Modal open={open} handleClose={handleClose} title=''>
      <Grid
        container
        justify='flex-start'
        spacing={3}
        direction='column'
        className={classes.root}
      >
        <Grid item container direction='column' alignItems='center'>
          <Icon className={classes.icon} src={exclamationMark} />
          <Typography variant='h3' className={classes.message}>
            Oops!
          </Typography>
        </Grid>
        <Grid item container direction='column'>
          <Grid item>
            <Typography variant='body2' className={classes.info}>
              Username didn't get registered. You are still anonymous.
            </Typography>
          </Grid>
          <Grid item container justify='center' alignItems='center'>
            <LoadingButton
              classes={{ button: classes.button }}
              text='Create a username'
              onClick={() => {
                openModalCreateUsername()
                handleClose()
              }}
            />
          </Grid>
        </Grid>
      </Grid>
    </Modal>
  )
}

FailedUsernameRegister.propTypes = {
  classes: PropTypes.object.isRequired,
  open: PropTypes.bool.isRequired,
  handleClose: PropTypes.func.isRequired,
  openModalCreateUsername: PropTypes.func.isRequired
}

FailedUsernameRegister.defaultProps = {
  open: false
}

export default R.compose(React.memo, withStyles(styles))(FailedUsernameRegister)
