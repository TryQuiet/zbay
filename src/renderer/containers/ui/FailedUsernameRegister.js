import React from 'react'
import { connect } from 'react-redux'
import { bindActionCreators } from 'redux'
import * as R from 'ramda'

import FailedUsernameRegister from '../../components/ui/FailedUsernameRegister'
import modalsHandlers, { withModal } from '../../store/handlers/modals'

export const mapStateToProps = state => ({})

export const mapDispatchToProps = dispatch =>
  bindActionCreators(
    {
      openModalCreateUsername: modalsHandlers.actionCreators.openModal('createUsernameModal')
    },

    dispatch
  )

export default R.compose(
  React.memo,
  connect(mapStateToProps, mapDispatchToProps),
  withModal('failedUsernameRegister')
)(FailedUsernameRegister)
