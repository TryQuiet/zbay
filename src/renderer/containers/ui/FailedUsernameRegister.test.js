/* eslint import/first: 0 */
jest.mock('../../vault')
import Immutable from 'immutable'

import { mapStateToProps, mapDispatchToProps } from './FailedUsernameRegister'

import create from '../../store/create'

describe('FailedUsernameRegister', () => {
  let store = null
  beforeEach(() => {
    jest.clearAllMocks()
    store = create({
      initialState: Immutable.Map({})
    })
  })

  it('will receive right props', async () => {
    const props = mapStateToProps(store.getState())
    expect(props).toMatchSnapshot()
  })

  it('will receive right actions', () => {
    const actions = mapDispatchToProps(x => x)
    expect(actions).toMatchSnapshot()
  })
})
