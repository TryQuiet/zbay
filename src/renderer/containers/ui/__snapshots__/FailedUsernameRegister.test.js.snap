// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FailedUsernameRegister will receive right actions 1`] = `
Object {
  "openModalCreateUsername": [Function],
}
`;

exports[`FailedUsernameRegister will receive right props 1`] = `Object {}`;
